pkg/controller/controller.go

```diff
@@ -186,20 +196,22 @@ func (c *Controller) clusterExists(name string, cl cloudprovider.Clusters) (bool
 
 // CreateComputePool create a compute node pool.
 func (c *Controller) CreateComputePool(p model.ComputePool) error {
-	cl, impl := c.Cloud.Clusters()
-	if !impl {
-		return ErrNotImplemented
-	}
 	pooler, impl := c.Cloud.NodePooler()
 	if !impl {
 		return ErrNotImplemented
 	}
 
-	// TODO Check if a masterpool for given cluster exists first? Maybe we
-	// don't care if masterpool does not exist yet, as long as the cluster infra exists?
-	if exists, err := c.clusterExists(p.ClusterName, cl); !exists {
+	clusters, err := c.GetClusters(p.ClusterName)
+	if err != nil {
 		return err
 	}
+	if len(clusters) != 1 {
+		return ErrClusterDoesNotExist
+	}
+	if len(clusters) > 1 {
+		return fmt.Errorf("more than one cluster found matching %q name", p.ClusterName)
```

Comment:
When `GetClusters` returns no results, this does not mean there is a problem. It's up to a caller to interpret that, hence the error above.